src/dbnode/storage/index/convert/convert_test.go

```diff
@@ -74,10 +75,44 @@ func TestFromSeriesIDAndTagsValid(t *testing.T) {
 	)
 	d, err := convert.FromSeriesIDAndTags(id, tags)
 	assert.NoError(t, err)
-	assert.Equal(t, "foo", string(d.ID))
-	assert.Len(t, d.Fields, 1)
-	assert.Equal(t, "bar", string(d.Fields[0].Name))
-	assert.Equal(t, "baz", string(d.Fields[0].Value))
+	assertContentsMatch(t, id, tags.Values(), d)
+}
+
+func TestFromSeriesIDAndTagsReuseBytesFromSeriesId(t *testing.T) {
+	tests := []struct {
+		name string
+		id   string
+	}{
+		{
+			name: "tags in ID",
+			id:   "bar=baz,quip=quix",
+		},
+		{
+			name: "tags in ID with specific format",
+			id:   `{bar="baz",quip="quix"}`,
+		},
+		{
+			name: "inexact tag occurrence in ID",
+			id:   "quixquip_bazillion_barometers",
+		},
+	}
+	tags := ident.NewTags(
+		ident.StringTag("bar", "baz"),
+		ident.StringTag("quip", "quix"),
+	)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			seriesID := ident.StringID(tt.id)
+			d, err := convert.FromSeriesIDAndTags(seriesID, tags)
+			assert.NoError(t, err)
+			assertContentsMatch(t, seriesID, tags.Values(), d)
+			for i := range d.Fields {
+				assert.True(t, test.ByteSliceDataContained(d.ID, d.Fields[i].Name))
+				assert.True(t, test.ByteSliceDataContained(d.ID, d.Fields[i].Value))
```

Comment:
Not 100% sure about this, my idea was to "find" the needed subslice within the id and then use the stock `ByteSlicesBackedBySameData` against it (then you wouldn't need to introduce this `ByteSliceDataContained`). But please decide yourself, that's just another possibility.

Reply:
https://github.com/m3db/m3/pull/3075/commits/6bb6d9b40510ae5404955846d12c5319a9dc0a9e#diff-4b7b04cd8028a198a21da071cd0a8bbcc9d2f7fd0f6f4bd896a73019eccdfdfeR306-R313